### PR TITLE
Remove broken Converter

### PR DESCRIPTION
**Pull request by SCullman**
Added a test that will fail for `Bar` ( would pass for `Foo 14`)

### CODE REVIEW

**Comment by SCullman** [comment]
@MangelMaxime, exception is thrown [here](https://github.com/thoth-org/Thoth.Json.Net/blob/07661f9eb4a7b7a4d753095cec3e4465701a4a40/src/Decode.fs#L100).
Can you please have a look?

Btw, it would not fail just for the decoder:
```fsharp
   type SimpleUnion = One | Two

   testCase "Auto.fromString works with simple Union" <| fun _ ->
       let json = "\"One\""
       Decode.Auto.fromString<SimpleUnion>(json, isCamelCase=true)
       |> equal (Ok One)
```

**Comment by SCullman** [comment]
Using the converter, decoding fails for `bool`, `string`, `float` and simple `union`s.
The same primitives won't fail if they are wrapped in any type.

**Comment by SCullman** [comment]
The converter is important for workflows that target Asp.Net WebApi within the .Net Framework.

**Comment by SCullman** [comment]
Good news:
`bool`, `string` and `float` are not failing anymore
Bad news:
`option string` is also failing, both `None` and `Some "text"`

**Comment by SCullman** [comment]
Yesterday I got the idea to get rid of the converter completely and created  [Thoth.Json.Net.Formatter](https://github.com/SCullman/Thoth.Json.Net.Formatter). First everything went smooth, and all unit tests for encoding and decoding were green.  I setup a test project for Asp:Net WebAPI, set it up. First GET request, nice. But all POSTs were never decoded back into parameters for my methods.

Today I went to plan C. I [rolled back](https://github.com/SCullman/DNN.Fable.Connector/tree/ThothJson2) to Thoth.Json 2.5.0, and patched Thoth.Fetch and Thoth.Json for Fable.Core 3 and Fable.Fetch 1.1.  First tests are promising, but  this doesn't feel good. 

In the current state this converter is just broken. @MangelMaxime , if you don't have an idea how to fix it I would recommend to remove it.

**Comment by SCullman** [comment]
I setup a new project [Fable.WebApi](https://github.com/SCullman/Fable.WebApi).

This project takes and transforms code and ideas from Thoth.Json(.Net) 2.5.0 within a new namespace `Djehuti.Json`. It supports now also Fable.Core 3. 

The idea is to offer a set of functions just to communicate between fable and Asp.Net WebApi using the AUTO De/Encoders and Newtonsoft.Json:

- A Fetch lib `Fable.WebApi.Fetch` inspired by Thoth.Fetch
- An attribute `FableWebApi` which can be applied to ApiControllers  or its methods to use the converter.

As I moved the old 2.5. code into new namespaces, developers can still use the newest Thoth.Json for safe access to third party services.

**Comment by SCullman** [comment]
### tl;dr
The converter did never work reliable in v.3.x. 
Its functionality can mostly be replaced with [Thoth.Json.Net.Formatter](https://github.com/SCullman/Thoth.Json.Net.Formatter)

**Comment by MangelMaxime** [comment]
Hello, @SCullman thank you for all your PRs.

I didn't forget you, but I have a delivery at work and Open F# workshop + talk to finish for the end of the month :)

And because your contributions needs time to think about, I prefer to wait for all that to pass instead of rushing your PRs.

**Comment by SCullman** [comment]
Thanks, that's good to know. I have created my workaround and therefore I am no longer in a hurry.

**Comment by MangelMaxime** [comment]
Hum, I am not sure to know what the difference between a converter and a formatter is. Neither do I know why the converter seems broken... In theory, if we pass it `System.Type` it should work the same as when the user call `Decode/Encode.Auto` unless we are losing some of the type information.

Any reason, why would someone use the converter/formatter instead of the "standard" API? 

I am not really a .Net dev so I prefer to ask questions :)

**Comment by SCullman** [comment]
What is a standard API? 

In .Net WebApi 2 (.Net Framework) a developer usually doesn't care about encoding/ decoding. 
It is just a class with custom methods. 
```csharp
[HttpGet]
public string Ping(){ return "Pong" }
```
The **client** can ask for different kind of serializations: plain text, XML, JSON. You can extend that list with _formatters_, for example to return the data as CSV. 
If the client asks for JSON, WebApi would serialize/deserialize with the help of a formatter provided by  Newtonsoft.Json. With _converters_ you can configure the serialization **within** Newtonsoft.Json, similiar to Thoth.Json extra coders. This worked pretty well with the build in converter in Thoth.Json 2.

Thoth.Json 3 did not provide any converter. I setup a PR for it, and managed to get the old unit test going green. But, a few month later, I realized that it is broken for a lot of use cases, added a few more tests and did not manage to get them green.

That's why I tried different solutions. One of them is the use of a _formatter_. 

There are different ways to setup formatters and converters. One way is to add attributes to your ApiControllers to tell WebApi why controllers should be used.

**Comment by MangelMaxime** [comment]
> What is a standard API?

By that, I meant `Decode.*` and `Encode.*`

Thank you for all the explanations.

> If the client asks for JSON, WebApi would serialize/deserialize with the help of a formatter provided by Newtonsoft.Json. With converters you can configure the serialization within Newtonsoft.Json, similiar to Thoth.Json extra coders. This worked pretty well with the build in converter in Thoth.Json 2.

Ok, I will need to have a look at why it doesn't work and add tests for it then.
